apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -15,7 +15,17 @@ defmodule Aecore.Chain.BlockValidation do
     chain_state_hash = ChainState.calculate_chain_state_hash(chain_state)
     is_valid_chain_state = ChainState.validate_chain_state(chain_state)
 
-    is_difficulty_target_met = Cuckoo.verify(new_block.header)
+    server = self()
+    work   = fn() -> Cuckoo.verify(new_block.header) end
+    spawn(fn() ->
+      send(server, {:worker_reply, self(), work.()})
+    end)
+
+    is_difficulty_target_met =
+      receive do
+      {:worker_reply, _from, verified?} -> verified?
+    end
```

Comment:
What's the point of adding this worker? We just execute Cuckoo.verify and immediately go to "receive do" where we wait for it to finish? How is this better then before?

Reply:
@d-velev please respond to my comment.

Reply:
@meivantodorov worked on this, he should be able to explain better

Reply:
@meivantodorov can you explain?

Reply:
Our cuckoo has own receive block and must be started as a worker. Otherwise it receives the parent's messages.